internal/app/referenceserver/impl.go

```diff
@@ -82,6 +79,42 @@ func (s *conformanceServer) Unary(
 	return resp, nil
 }
 
+func (s *conformanceServer) IdempotentUnary(
+	ctx context.Context,
+	req *connect.Request[v1.IdempotentUnaryRequest],
+) (*connect.Response[v1.IdempotentUnaryResponse], error) {
+	if req.HTTPMethod() != http.MethodGet {
+		return nil, errors.New("idempotent unary method should have been called via HTTP GET")
+	}
```

Comment:
FWIW, this should already be checked by the reference server in `checks.go`.

This should be pretty straight-forward to share code with `Unary`, right? Maybe at least a TODO?

Reply:
Yeah, but it still involves the same issue as the client -- the different request/response type. I added a TODO to handle this the same way.

Reply:
> the different request/response type

This case is much simpler though: you just don't put that part in the shared function. Have the shared function accept the parts of the request (`requestData []byte` and `*UnaryResponseDefinition`) and return the `*ConformancePayload`. Then the only thing these two RPC methods do is extract those two parameters, call the shared function, and then package the payload in the right response type.

> I added a TODO to handle this the same way.

Cool, that's fine.